core/server/worker/src/main/java/alluxio/worker/block/BlockHeartbeatReporter.java

```diff
@@ -139,13 +142,13 @@ private void removeBlockInternal(long blockId) {
    * Adds a block to the list of added blocks in this heartbeat period.
    *
    * @param blockId the id of the block to add
-   * @param tierAlias alias of the storage tier containing the block
+   * @param location BlockStoreLocation containing the blockid
    */
-  private void addBlockToAddedBlocks(long blockId, String tierAlias) {
-    if (mAddedBlocks.containsKey(tierAlias)) {
-      mAddedBlocks.get(tierAlias).add(blockId);
+  private void addBlockToAddedBlocks(long blockId, BlockStoreLocation location) {
+    if (mAddedBlocks.containsKey(location)) {
+      mAddedBlocks.get(location).add(blockId);
     } else {
-      mAddedBlocks.put(tierAlias, Lists.newArrayList(blockId));
+      mAddedBlocks.put(location, Lists.newArrayList(blockId));
```

Comment:
Does this work because the hash will be different for different dirs but we need to aggregate across dirs when we report to the master?

Reply:
Good catch, this was from before when we passed information keyed by tieralias/dirIndex/storageMedium to the master.